File: app/api/route/dept.go
package route

import (
	"github.com/gin-gonic/gin"
	"github.com/hulutech-web/workflow-engine/app/api/schemas/req"
	"github.com/hulutech-web/workflow-engine/app/api/service"
	"github.com/hulutech-web/workflow-engine/app/api/types"
	"github.com/hulutech-web/workflow-engine/app/models"
	"github.com/hulutech-web/workflow-engine/pkg/plugin/response"
	"github.com/sirupsen/logrus"
	"github.com/spf13/cast"
	"go.uber.org/fx"
	"net/http"
)

type dept struct {
	fx.In
	Srv service.DeptService
}

func deptRoutes(a dept, r *types.ApiRouter) {
	r.POST("/dept", a.Store)
	r.PUT("/dept", a.Update)
	r.GET("/dept", a.Index)
	r.DELETE("/dept/:id", a.Destroy)
	r.GET("/dept/:id", a.Show)
	r.GET("/dept/list", a.List)
	r.POST("/dept/bind_manager", a.BindManager)
	r.POST("/dept/bind_director", a.BindDirector)
	r.GET("/dept/:id/tree", a.DisplayTree)
}

// Index @BasePath /api
// @Summary 部门
// @Description 部门分页
// @Tags Dept 部门
// @Id DeptIndex
// @Produce json
// @Success 200 {object} service.PageResult "成功"
// @Router /dept [get]
func (r *dept) Index(ctx *gin.Context) {
	query := ctx.Request.URL.Query()
	index, err := r.Srv.Index(ctx, query)
	if err != nil {
		response.Fail(ctx, response.Failed)
		return
	}
	logrus.WithFields(logrus.Fields{
		"index": index,
	}).Info("返回成功")
	response.OkWithData(ctx, index)
}

// List @BasePath /api
// @Summary 部门
// @Description 部门列表
// @Tags Dept 部门
// @Id DeptList
// @Produce json
// @Success 200 {object} response.Response{data=map[string]interface{}} "成功"
// @Router /dept/list [get]
func (r *dept) List(ctx *gin.Context) {
	list, err := r.Srv.List(ctx)
	if err != nil {
		response.Fail(ctx, response.Failed)
	}
	response.OkWithData(ctx, list)
}

// Show @BasePath /api
// @Summary 部门
// @Description 单个部门
// @Tags Dept 部门
// @Id DeptShow
// @Produce json
// @Param id path int true "部门ID"
// @Success 200 {object} response.Response{data=map[string]interface{}} "成功"
// @Router /dept/{id} [get]
func (r *dept) Show(ctx *gin.Context) {
	id := ctx.Param("id")
	idInt := cast.ToInt(id)
	show := r.Srv.Show(ctx, idInt)
	response.OkWithData(ctx, show)
}

// Store @BasePath /api
// @Summary 部门
// @Description 新增部门
// @Tags Dept 部门
// @Id DeptStore
// @Produce json
// @Param request body models.Dept true "部门信息"
// @Success 200 {object} response.Response{data=map[string]interface{}} "成功"
// @Router /dept [post]
func (r *dept) Store(ctx *gin.Context) {
	var dpt models.Dept
	ctx.Bind(&dpt)
	err := r.Srv.Store(ctx, dpt)
	if err != nil {
		response.FailWithMsg(ctx, response.Failed, err.Error())
		return
	}
	response.OkWithData(ctx, "操作成功")
}

// Update @BasePath /api
// @Summary 部门
// @Description 新增部门
// @Tags Dept 部门
// @Id DeptUpdate
// @Produce json
// @Param id path int true "部门ID"
// @Param request body models.Dept true "部门信息"
// @Success 200 {object} response.Response{data=map[string]interface{}} "成功"
// @Router /dept/{id} [put]
func (r *dept) Update(ctx *gin.Context) {
	var dpt models.Dept
	ctx.Bind(&dpt)
	err := r.Srv.Update(ctx, dpt)
	if err != nil {
		response.FailWithMsg(ctx, response.Failed, err.Error())
		return
	}
	response.OkWithData(ctx, "操作成功")
}

// Destroy @BasePath /api
// @Summary 部门
// @Description 新增部门
// @Tags Dept 部门
// @Id DeptDestroy
// @Produce json
// @Param id path int true "部门ID"
// @Success 200 {object} response.Response{data=map[string]interface{}} "成功"
// @Router /dept/{id} [delete]
func (r *dept) Destroy(ctx *gin.Context) {
	id := ctx.Param("id")
	err := r.Srv.Destroy(ctx, cast.ToInt(id))
	if err != nil {
		response.Fail(ctx, response.Failed)
		return
	}
	response.OkWithData(ctx, "操作成功")
}

// BindManager @BasePath /api
// @Summary 部门
// @Description 新增部门
// @Tags Dept 部门
// @Id DeptBindManager
// @Produce json
// @Param request body req.BindManagerReq true "绑定参数"
// @Success 200 {object} response.Response{data=map[string]interface{}} "成功"
// @Router /dept/bind_manager [post]
func (r *dept) BindManager(ctx *gin.Context) {
	var bindManagerReq req.BindManagerReq
	if err2 := ctx.ShouldBindJSON(&bindManagerReq); err2 != nil {
		response.FailWithMsg(ctx, response.Failed, err2.Error())
		return
	}

	err := r.Srv.BindManager(ctx, bindManagerReq.ManagerID, bindManagerReq.DeptID)
	if err != nil {
		response.Fail(ctx, response.Failed)
		return
	}
	response.OkWithData(ctx, "操作成功")
}

// BindDirector @BasePath /api
// @Summary 部门
// @Description 新增部门
// @Tags Dept 部门
// @Id DeptBindDirector
// @Produce json
// @Param request body req.BindDirectorReq true "绑定参数"
// @Success 200 {object} response.Response{data=map[string]interface{}} "成功"
// @Router /dept/bind_director [post]
func (r *dept) BindDirector(ctx *gin.Context) {
	var bindDirectorReq req.BindDirectorReq
	if err2 := ctx.ShouldBindJSON(&bindDirectorReq); err2 != nil {
		response.FailWithMsg(ctx, response.Failed, err2.Error())
		return
	}

	err := r.Srv.BindDirector(ctx, bindDirectorReq.DirectorID, bindDirectorReq.DeptID)
	if err != nil {
		ctx.JSON(http.StatusOK, gin.H{
			"error": err.Error(),
		})
		return
	}
	response.OkWithData(ctx, "操作成功")
}

// DisplayTree @BasePath /api
// @Summary 部门
// @Description 新增部门
// @Tags Dept 部门
// @Id DeptDisplayTree
// @Param id path int true "部门ID"
// @Success 200 {object} response.Response{data=map[string]interface{}} "成功"
// @Router /dept/{id}/tree [get]
func (r *dept) DisplayTree(ctx *gin.Context) {
	id := ctx.Param("id")
	res, err := r.Srv.DisplayTree(ctx, cast.ToInt(id))
	if err != nil {
		response.Fail(ctx, response.Failed)
		return
	}
	response.OkWithData(ctx, res)
}

File: front-end/src/api/dept.ts
// @ts-ignore
/* eslint-disable */
import request from '@/utils/request/http';

/** 部门 部门分页 GET /dept */
export async function deptIndex(options?: RequestOptions) {
  return request<API.PageResult>('/dept', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 部门 新增部门 POST /dept */
export async function deptStore(body: API.Dept, options?: RequestOptions) {
  return request<API.Response & { data?: Record<string, any> }>('/dept', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 部门 单个部门 GET /dept/${param0} */
export async function deptShow(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.DeptShowParams,
  options?: RequestOptions
) {
  const { id: param0, ...queryParams } = params;
  return request<API.Response & { data?: Record<string, any> }>(`/dept/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 部门 新增部门 PUT /dept/${param0} */
export async function deptUpdate(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.DeptUpdateParams,
  body: API.Dept,
  options?: RequestOptions
) {
  const { id: param0, ...queryParams } = params;
  return request<API.Response & { data?: Record<string, any> }>(`/dept/${param0}`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    params: { ...queryParams },
    data: body,
    ...(options || {}),
  });
}

/** 部门 新增部门 DELETE /dept/${param0} */
export async function deptDestroy(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.DeptDestroyParams,
  options?: RequestOptions
) {
  const { id: param0, ...queryParams } = params;
  return request<API.Response & { data?: Record<string, any> }>(`/dept/${param0}`, {
    method: 'DELETE',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 部门 新增部门 GET /dept/${param0}/tree */
export async function deptDisplayTree(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.DeptDisplayTreeParams,
  options?: RequestOptions
) {
  const { id: param0, ...queryParams } = params;
  return request<API.Response & { data?: Record<string, any> }>(`/dept/${param0}/tree`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 部门 新增部门 POST /dept/bind_director */
export async function deptBindDirector(body: API.BindDirectorReq, options?: RequestOptions) {
  return request<API.Response & { data?: Record<string, any> }>('/dept/bind_director', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 部门 新增部门 POST /dept/bind_manager */
export async function deptBindManager(body: API.BindManagerReq, options?: RequestOptions) {
  return request<API.Response & { data?: Record<string, any> }>('/dept/bind_manager', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 部门 部门列表 GET /dept/list */
export async function deptList(options?: RequestOptions) {
  return request<API.Response & { data?: Record<string, any> }>('/dept/list', {
    method: 'GET',
    ...(options || {}),
  });
}
